RHEL6 - CentOS (8) TestDockerDefFile failures (kernel too old)

### Version of Singularity:

```
3.5.2
```

### Actual behavior

`integration-test` failures as RHEL6 kernel can't run latest CentOS8 containes:

```
--- FAIL: TestDockerDefFile (44.00s)
    --- SKIP: TestDockerDefFile/Arch (0.00s)
        docker_test.go:189: kernel >=3.x required
    --- PASS: TestDockerDefFile/BusyBox (4.87s)
        --- PASS: TestDockerDefFile/BusyBox/False (0.08s)
        --- PASS: TestDockerDefFile/BusyBox/RunScript (0.08s)
        --- PASS: TestDockerDefFile/BusyBox/OneBase (0.09s)
        --- PASS: TestDockerDefFile/BusyBox/ActionsShell (0.08s)
        --- PASS: TestDockerDefFile/BusyBox/ActionsExec (0.08s)
        --- PASS: TestDockerDefFile/BusyBox/ActionsRun (0.13s)
        --- PASS: TestDockerDefFile/BusyBox/Environment (0.09s)
        --- PASS: TestDockerDefFile/BusyBox/Singularity (0.08s)
    --- FAIL: TestDockerDefFile/CentOS (22.98s)
        --- PASS: TestDockerDefFile/CentOS/False (0.10s)
        --- FAIL: TestDockerDefFile/CentOS/RunScript (0.09s)
            build_test.go:116: FATAL: kernel too old
                
            build_test.go:117: unexpected failure running 'test -f /.singularity.d/runscript': <nil>
        --- FAIL: TestDockerDefFile/CentOS/OneBase (0.09s)
            build_test.go:116: FATAL: kernel too old
                
            build_test.go:117: unexpected failure running 'test -f /.singularity.d/env/01-base.sh': <nil>
        --- FAIL: TestDockerDefFile/CentOS/ActionsShell (0.08s)
            build_test.go:116: FATAL: kernel too old
                
            build_test.go:117: unexpected failure running 'test -f /.singularity.d/actions/shell': <nil>
        --- FAIL: TestDockerDefFile/CentOS/ActionsExec (0.09s)
            build_test.go:116: FATAL: kernel too old
                
            build_test.go:117: unexpected failure running 'test -f /.singularity.d/actions/exec': <nil>
        --- FAIL: TestDockerDefFile/CentOS/ActionsRun (0.11s)
            build_test.go:116: FATAL: kernel too old
                
            build_test.go:117: unexpected failure running 'test -f /.singularity.d/actions/run': <nil>
        --- FAIL: TestDockerDefFile/CentOS/Environment (0.07s)
            build_test.go:116: FATAL: kernel too old
                
            build_test.go:117: unexpected failure running 'test -L /environment': <nil>
        --- FAIL: TestDockerDefFile/CentOS/Singularity (0.08s)
            build_test.go:116: FATAL: kernel too old
                
            build_test.go:117: unexpected failure running 'test -L /singularity': <nil>
```

### What OS/distro are you running

```
RHEL 6
```

--- a/cmd/singularity/docker_test.go
+++ b/cmd/singularity/docker_test.go
@@ -176,8 +176,11 @@ func TestDockerDefFile(t *testing.T) {
 	}{
 		{"Arch", 3, "dock0/arch:latest"},
 		{"BusyBox", 0, "busybox:latest"},
-		{"CentOS", 0, "centos:latest"},
-		{"Ubuntu", 0, "ubuntu:16.04"},
+		{"CentOS_6", 0, "centos:6"},
+		{"CentOS_7", 0, "centos:7"},
+		{"CentOS_8", 3, "centos:8"},
+		{"Ubuntu_1604", 0, "ubuntu:16.04"},
+		{"Ubuntu_1804", 3, "ubuntu:18.04"},
 	}
 
 	for _, tt := range tests {

--- a/e2e/docker/docker.go
+++ b/e2e/docker/docker.go
@@ -268,14 +268,29 @@ func (c ctx) testDockerDefFile(t *testing.T) {
 			from:                "busybox:latest",
 		},
 		{
-			name:                "CentOS",
+			name:                "CentOS_6",
 			kernelMajorRequired: 0,
-			from:                "centos:latest",
+			from:                "centos:6",
 		},
 		{
-			name:                "Ubuntu",
+			name:                "CentOS_7",
+			kernelMajorRequired: 0,
+			from:                "centos:7",
+		},
+		{
+			name:                "CentOS_8",
+			kernelMajorRequired: 3,
+			from:                "centos:8",
+		},
+		{
+			name:                "Ubuntu_1604",
 			kernelMajorRequired: 0,
 			from:                "ubuntu:16.04",
+		},
+		{
+			name:                "Ubuntu_1804",
+			kernelMajorRequired: 3,
+			from:                "ubuntu:18.04",
 		},
 	}
 

